### PR TITLE
Update to SPP v5

### PR DESCRIPTION
**Pull request by MrGenga**
### Description

Adding all updates of SynapseAPI, to work with Nemisys.
### Reason to modify

I wan't to more stable working of my sevrer.

### Tests & Reviews

Tested on my own server. Works fine, Nemisys working. There is no Synapse crashes in console.
Based on SynapseAPI, contains packet system update, update to SPP v4 and then SPP v5.

### CODE REVIEW

**Comment by ishitatsuyuki** [comment]
Edited on GitHub: something proves that you haven't tested the thing!

**Comment by MrGenga** [comment]
Fixed. Works fine.

**Comment by MrGenga** [comment]
Approved! Looks good.

**Comment by dktapps** [comment]
Far too much commit spam. Please squash your commits.

**Comment by flaxues** [comment]
It works for me

**Comment by jojoe77777** [comment]
> Fixed. Works fine.

`$pk->direct = direct;`
`$this->sendDataPacket(pk);`
This isn't Java...

**Comment by ishitatsuyuki** [comment]
It's clear that you haven't tested this. If you want to ensure code quality, editing online is the worst way.

**Comment by MrGenga** [comment]
Now i have to dev this on my VDS!

File: src/synapse/network/synlib/ServerConnection.php [hunk 1]
```diff
@@ -155,6 +155,8 @@ public function readPacket(){
                 $this->receiveBuffer = "";
              }
 		}
+
+                $this->receiveBuffer = ""; //Now I know - eeceiveBuffer is empty
```
Comment (dktapps) [inline src/synapse/network/synlib/ServerConnection.php]: Your formatting is a horrible mess. Please use tabs, not spaces.

**Comment by dktapps** [comment]
Formatting issues, use of GitHub web editor, lots of amendments since opening the PR, far too much commit spam. Code Review -1.

**Comment by dktapps** [comment]
It seems to me that you don't have a clear idea what you are doing, and you're making an awful mess of this. I think somebody else who _does_ know what they are doing should take up this job. Closing.

**Comment by MrGenga** [comment]
@dktapps WHY!? I trying to clear memory, because SynLib crashs! Reopen!

**Comment by ishitatsuyuki** [comment]
Rule 1: Done your work before submitting.
Rule 2: Squash your commits, keep the message clean.
Rule 3: Do not use web editor for big changes, it's identical to say that "I AM A IDIOT!"

**Pull request closed without merging**